File: server/src/main/java/me/ajfleming/tworoomsio/socket/response/Response.java
package me.ajfleming.tworoomsio.socket.response;

import lombok.Value;

@Value
public class Response {

  boolean success;
  String message;

  public static Response success(final String message) {
    return new Response(true, message);
  }

  public static Response error(final String message) {
    return new Response(false, message);
  }
}

File: kotlin_server/src/main/kotlin/me/ajfleming/tworoomsio/socket/response/Response.kt
package me.ajfleming.tworoomsio.socket.response

class Response (
    val success: Boolean,
    val message: String? = null
) {
    companion object {
        fun success(message: String?): Response {
            return Response(true, message)
        }

        fun error(message: String?): Response {
            return Response(false, message)
        }
    }
}